### DISCUSSION TITLE
DIAG Pin assignment

### DISCUSSION THREAD

**Discussion by asparagustuss** [Chr157i4n/PyTmcStepper#73] [General]
Really great stuff here. Just curious why the diag pin isn't part of the driver initialization. I'm tempted to change the library, but I was curious to your thoughts as to why its left outside of the scope.

**Comment by Chr157i4n** [comment]
Thanks for the feedback.

I honestly haven't given it much thought...

The only real use for the diag pin is StallGuard, isnt it?

Currently you have to setup StallGuard with your callback function like this
```python
tmc.set_stallguard_callback(pin_stallguard, threshold, callback)
```

But yes, with keyword arguments in the constructor, you could also setup StallGuard there. If you want, you are welcome to make a pull request.
